src/header/common/link.rs

```diff
@@ -0,0 +1,622 @@
+use std::fmt;
+use std::str::FromStr;
+
+use uri::Uri;
+use mime::Mime;
+use language_tags::LanguageTag;
+
+use header::{Header, Raw};
+use header::parsing::from_one_raw_str;
+
+/// The `Link` header, defined in
+/// [RFC5988](http://tools.ietf.org/html/rfc5988#section-5)
+///
+/// # ABNF
+/// ```plain
+/// Link           = "Link" ":" #link-value
+/// link-value     = "<" URI-Reference ">" *( ";" link-param )
+/// link-param     = ( ( "rel" "=" relation-types )
+///                | ( "anchor" "=" <"> URI-Reference <"> )
+///                | ( "rev" "=" relation-types )
+///                | ( "hreflang" "=" Language-Tag )
+///                | ( "media" "=" ( MediaDesc | ( <"> MediaDesc <"> ) ) )
+///                | ( "title" "=" quoted-string )
+///                | ( "title*" "=" ext-value )
+///                | ( "type" "=" ( media-type | quoted-mt ) )
+///                | ( link-extension ) )
+/// link-extension = ( parmname [ "=" ( ptoken | quoted-string ) ] )
+///                | ( ext-name-star "=" ext-value )
+/// ext-name-star  = parmname "*" ; reserved for RFC2231-profiled
+/// ; extensions.  Whitespace NOT
+/// ; allowed in between.
+/// ptoken         = 1*ptokenchar
+/// ptokenchar     = "!" | "#" | "$" | "%" | "&" | "'" | "("
+///                | ")" | "*" | "+" | "-" | "." | "/" | DIGIT
+///                | ":" | "<" | "=" | ">" | "?" | "@" | ALPHA
+///                | "[" | "]" | "^" | "_" | "`" | "{" | "|"
+///                | "}" | "~"
+/// media-type     = type-name "/" subtype-name
+/// quoted-mt      = <"> media-type <">
+/// relation-types = relation-type
+///                | <"> relation-type *( 1*SP relation-type ) <">
+/// relation-type  = reg-rel-type | ext-rel-type
+/// reg-rel-type   = LOALPHA *( LOALPHA | DIGIT | "." | "-" )
+/// ext-rel-type   = URI
+/// ```
+///
+/// # Example values
+///
+/// ...
+///
+/// # Examples
+///
+/// ...
+///
+///
+#[derive(Clone, PartialEq, Debug)]
+pub struct Link {
+    /// Target IRI: `link-value`
+    pub link: Uri,
```

Comment:
A builder may be a good idea!

I didn't mean to make those 2 fields generic, but rather, I was wondering if they are served well enough as `String`, and if someone wanted to, they can make a whole new `struct MyLink { ... }` header.